src/ray/core_worker/experimental_channel_manager.cc

```diff
@@ -0,0 +1,200 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// PLASMA CLIENT: Client library for using the plasma store and manager
+
+#include "ray/core_worker/experimental_channel_manager.h"
+
+namespace ray {
+
+Status ExperimentalChannelManager::RegisterWriterChannel(
+    const ObjectID &object_id, std::unique_ptr<plasma::MutableObject> mutable_object) {
+#ifdef __linux__
```

Comment:
Q: Do we raise an exception if this feature is used in MacOS now?

Reply:
No.